### DISCUSSION TITLE
Devise omniauthable

### DISCUSSION THREAD

**Discussion by maricavor** [graphql-devise/graphql_devise#252] [General]
How to integrate omniauth google or facebook login?

**Comment by 00dav00** [comment]
Hi @maricavor , I think the question is a little broad so I would like to collaborate my 2 cents since I'm not really sure of the use case.

- OAuth is heavily reliant on redirects, GQL responses always use a 200 HTTP code, which can be worked around but kind of tells you this is flow is probably thought with REST endpoints in mind
- Checking the [omniauth gem](https://github.com/zquestz/omniauth-google-oauth2#devise) I see the devise integration requires some [callback endpoints](https://github.com/heartcombo/devise/blob/main/app/controllers/devise/omniauth_callbacks_controller.rb) to be defined. This probably means that the auth provider itself is expecting to call a REST endpoint.

I think that REST endpoints are probably a good fit if you are planning to use OAuth. WDYT?

**Reply by maricavor** [reply]
Hi, @00dav00!  Thanks for your reply. I use graphql_devise for my API. I started using Facebook and Google OAuth Login, but on frontend side where I have defined a callback endpoint. Then I created separate mutation for a receiving only access token from this callback on backend side where I use graphql_devise. This mutation is responsible for finding facebook or google account data by access_token, logging in the user and sending response to frontend with the userinfo.
